tests/queries/0_stateless/02703_row_policy_for_database.sql

```diff
@@ -0,0 +1,53 @@
+-- Tags: no-parallel
+
+DROP DATABASE IF EXISTS db1_02703;
+DROP USER IF EXISTS u1_02703;
+CREATE USER u1_02703;
+
+CREATE DATABASE db1_02703;
+
+CREATE TABLE db1_02703.02703_rqtable (x UInt8) ENGINE = MergeTree ORDER BY x;
+INSERT INTO db1_02703.02703_rqtable VALUES (1), (2), (3), (4);
+
+
+SELECT '-- row policies for database';
+CREATE ROW POLICY db1_02703 ON db1_02703.* USING 1 AS PERMISSIVE TO ALL;
+CREATE ROW POLICY tbl1_02703 ON db1_02703.table USING 1 AS PERMISSIVE TO ALL;
+SELECT '  -- SHOW CREATE POLICY db1_02703 ON db1_02703.*';
+SHOW CREATE POLICY db1_02703 ON db1_02703.*;
+SELECT '  -- SHOW CREATE POLICY ON db1_02703.*';
+SHOW CREATE POLICY ON db1_02703.*;
+SELECT '  -- SHOW CREATE POLICY ON db1_02703.`*`';
+SHOW CREATE POLICY ON db1_02703.`*`;
+DROP POLICY db1_02703 ON db1_02703.*;
+DROP POLICY tbl1_02703 ON db1_02703.table;
+
+CREATE ROW POLICY any_02703 ON *.some_table USING 1 AS PERMISSIVE TO ALL; -- { clientError 62 }
+
+CREATE TABLE 02703_rqtable_default (x UInt8) ENGINE = MergeTree ORDER BY x;
+
+CREATE ROW POLICY 02703_filter_11_db ON * USING x=1 AS permissive TO ALL;
+CREATE ROW POLICY 02703_filter_11 ON 02703_rqtable_default USING x=2 AS permissive TO ALL;
+
+INSERT INTO 02703_rqtable_default VALUES (1), (2), (3), (4);
+
+SELECT 'R1, R2: (x == 1) OR (x == 2)';
+SELECT * FROM 02703_rqtable_default;
+
+DROP TABLE 02703_rqtable_default;
+
+SELECT 'Check system.query_log';
+SYSTEM FLUSH LOGS;
+SELECT query, used_row_policies FROM system.query_log WHERE current_database == currentDatabase() AND type == 'QueryStart' AND query_kind == 'Select' ORDER BY event_time_microseconds;
```

Comment:
You can try to debug it by replacing this line temporarily with

```
SELECT query, used_row_policies, type, query_kind, event_time_microseconds FROM system.query_log WHERE current_database == currentDatabase();
```

Reply:
Hello @vitibar, the thing is 02131_used_row_policies_in_query_log.sql does not work either if allow_experimental_analyzer=1 with exactly the same symptoms

```
-SELECT * FROM 02131_rqtable;	['`02131_filter_4` ON default.`02131_rqtable`','`02131_filter_5` ON default.`02131_rqtable`']
+SELECT * FROM 02131_rqtable;	[]
 SELECT \'R5: (x >= 2)\';	[]
-SELECT * FROM 02131_rqtable;	['`02131_filter_5` ON default.`02131_rqtable`']
+SELECT * FROM 02131_rqtable;	[]

```
No row level policies in query_log regardless of type and query_kind.
I presume that it is a known analyzer limitation, isn't it?

Reply:
As far as I can see the test 02131_used_row_policies_in_query_log is passed for analyzer because it is listed in tests/broken_tests.txt
Looks weird.
Especially because the list is general, not analyzer specific.

Reply:
Yes, you're right, so probably you can just add your test to that broken list.
The file `tests/broken_tests.txt` only affects the tests with analyzer.

Reply:
https://github.com/ClickHouse/ClickHouse/blob/master/tests/ci/functional_test_check.py#L110